### PR TITLE
Implement groups model and update index.html accordingly

### PR DESCRIPTION
**Pull request by Sasafrass**
People can now be added to groups manually using database queries. 
Index.html also prints all the groups your user is a member of.
Flask SQL-Alchemy doesn't like the way I implemented the table, so it raises a small error once in the command line.
This PR comes with a migration script, so upon updating you may wish to `flask db upgrade` in your terminal to upgrade to the latest version of the database.

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -3,12 +3,27 @@
...
     username = db.Column(db.String(64), index=True, unique=True)
     email = db.Column(db.String(120), index=True, unique=True)
     password_hash = db.Column(db.String(128))
     words = db.relationship("Slang", backref="author", lazy="dynamic")
+    groups = db.relationship(
+        "Group", 
+        secondary=groups, 
+        lazy="subquery",
+        backref=db.backref("groups", lazy=True),
+        overlaps="users, groups",
```
Comment (Sasafrass) [inline app/models.py:25]: I've been trying different permutations of this "overlaps=." parameter to suppress the CMD warning by flask upon navigation to index.html when logged in but to no avail

File: app/models.py [hunk 2]
```diff
@@ -30,6 +45,18 @@ def __repr__(self):
...
 
+class Group(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(128), index=True, unique=True)
+    users = db.relationship(
+        "User", 
+        secondary=groups, 
+        lazy="subquery",
+        backref=db.backref("users", lazy=True),
+        overlaps="groups, users",
```
Comment (Sasafrass) [inline app/models.py:56]: See comment at line 25

**Comment by raoulg** [comment]
Well, for what I see, it seems to add a line `Your current groups are []`. 
But my knowledge of flask is really to premature to say anything beyond that it does not crash on my machine...

**Comment by Sasafrass** [comment]
> Well, for what I see, it seems to add a line `Your current groups are []`.
> But my knowledge of flask is really to premature to say anything beyond that it does not crash on my machine...

Yeah, it works but it just gives a random flask error in the CLI

**Comment by Sasafrass** [comment]
Should now be possible to create a group (you must be logged in to do so). No additional functionality though, so it also doesn't display your updated group when you print all of your goups.
The "files changed" has become a bit of a mess since commit 8d2233d59d512dd14940765103b8adfd8172bec6 which also did a ton of linting, so you'd better check out 8d2233d59d512dd14940765103b8adfd8172bec6

File: app/ml_models/rnn/generate.py [hunk 3]
```diff
@@ -48,6 +48,7 @@ def generate_word(
             .unsqueeze(0)
             .to(device)
         )
+
```
Comment (annaproxy) [inline app/ml_models/rnn/generate.py:51]: love it

**Review by annaproxy — APPROVED** [review]

**Pull request merged**